roachpb/data_test.go

```diff
@@ -695,3 +697,48 @@ func TestTransactionIDLen(t *testing.T) {
 		t.Fatalf("expected %d, got %d", TransactionIDLen, l)
 	}
 }
+
+func findPtrFields(t reflect.Type, prefix string) []string {
+	if t.Kind() != reflect.Struct {
+		return nil
+	}
+	var res []string
+	for i := 0; i < t.NumField(); i++ {
+		f := t.Field(i)
+		switch f.Type.Kind() {
+		case reflect.Ptr:
+			res = append(res, prefix+f.Name)
+			if f.Type.Elem().Kind() == reflect.Struct {
+				res = append(res, findPtrFields(f.Type.Elem(), f.Name+".")...)
+			}
+		case reflect.Slice:
+			res = append(res, prefix+f.Name)
+			res = append(res, findPtrFields(f.Type.Elem(), f.Name+".")...)
+		case reflect.Struct:
+			res = append(res, findPtrFields(f.Type, f.Name+".")...)
+		}
+	}
+	return res
+}
+
+func TestTransactionClone(t *testing.T) {
```

Comment:
I'm not quite satisfied with this test yet. I want a test that breaks if additional pointer fields are added to `Transaction`. This is achieved below by keeping a whitelist of the current pointer fields. Still thinking about whether I can do something better.